### PR TITLE
Don't panic when body element don't have style data

### PR DESCRIPTION
**Pull request by upsuper**
This is the Servo side change of [bug 1381233](https://bugzilla.mozilla.org/show_bug.cgi?id=1381233).

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17799)

### CODE REVIEW

File: components/style/values/specified/color.rs [hunk 1]
```diff
@@ -269,16 +269,12 @@ impl ToComputedValue for Color {
...
-                    Gecko_GetBody(pres_context)
-                };
-                if let Some(body) = body {
-                    let wrap = GeckoElement(body);
-                    let borrow = wrap.borrow_data();
-                    ComputedColor::rgba(borrow.as_ref().unwrap()
-                                              .styles.primary()
-                                              .get_color()
-                                              .clone_color())
+                let body = unsafe { Gecko_GetBody(pres_context) }.map(|body| GeckoElement(body));
```
Comment (emilio) [inline components/style/values/specified/color.rs]: nit: I think you can just use `map(GeckoElement)`

**Comment by upsuper** [comment]
@bors-servo r-

Let me address emilio's comment.

**Pull request merged**